lib/src/over_react_redux/devtools/middleware.dart

```diff
@@ -0,0 +1,158 @@
+@JS()
+library over_react_redux.devtools.middleware;
+
+import 'dart:convert';
+import 'dart:html';
+import 'dart:js_util';
+
+import 'package:logging/logging.dart';
+import 'package:react/react_client/js_backed_map.dart';
+import 'package:redux/redux.dart';
+import 'package:redux_dev_tools/redux_dev_tools.dart';
+import 'package:js/js.dart';
+
+@JS()
+@anonymous
+class ReduxDevToolsExtensionConnection {
+  external factory ReduxDevToolsExtensionConnection();
+  external subscribe(dynamic Function(JsMap data) listener);
+  external send([dynamic action, dynamic state]);
+  external init([dynamic initialState]);
+}
+
+@JS('__REDUX_DEVTOOLS_EXTENSION__.connect')
+external ReduxDevToolsExtensionConnection reduxExtConnect([dynamic options]);
+
+class OverReactReduxDevToolsMiddleware extends MiddlewareClass {
+  Store _store;
+  ReduxDevToolsExtensionConnection devToolsExt;
+  final Logger log = Logger('OverReactReduxDevToolsMiddleware');
+
+  OverReactReduxDevToolsMiddleware() {
+    log.onRecord.listen((rec) {
+      if (rec.level == Level.WARNING) {
+        window.console.warn('${log.name} [${rec.level.name}]: ${rec.message}');
+      } else {
+        window.console.log('${log.name} [${rec.level.name}]: ${rec.message}');
+      }
+    });
+    try {
+      devToolsExt = reduxExtConnect();
+    } catch(_) {
+      log.warning(
+        'Unable to connect to the redux dev tools browser extension.\n'
+        'Please install it...\n'
+        'Chrome: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en\n'
+        'Firefox: https://addons.mozilla.org/en-US/firefox/addon/reduxdevtools/\n'
+      );
+    }
+  }
+
+  set store(Store v) {
+    _store = v;
+    devToolsExt.init(_encodeForTransit(store.state));
+    devToolsExt.subscribe(handleEventFromRemote);
+  }
+
+  get store => _store;
+
+  dynamic _encodeForTransit(dynamic content, {bool shouldRethrow = false}) {
+    try {
+      return jsify(jsonDecode(jsonEncode(content)));
+    } catch (_) {
+      log.warning('You must implement a `toJson` method in your state and actions in order to view state changes in the redux dev tools.');
+      if (shouldRethrow) rethrow;
+    }
+  }
+
+  dynamic _encodeActionForTransit(dynamic action) {
+    try {
+      return _encodeForTransit({'type': _getActionType(action), 'payload': action}, shouldRethrow: true);
+    } on Error {
+      return _encodeForTransit({'type': _getActionType(action)});
+    }
+  }
+
+  String _getActionType(dynamic action) {
+    if (action.toString().contains('Instance of')) {
```

Comment:
#nit this is a more specific check
```suggestion
    if (action.toString().startsWith('Instance of')) {
```